table_foosball/table_foosball_test.rb

```diff
@@ -9,50 +9,41 @@ def setup
     @kenichi = Player.new("Kenichi")
   end
 
+  def test_create_new_player
+    frederik = Player.new("Frederik")
+    assert_equal "Frederik", frederik.name
+    assert_empty frederik.games
+  end
+
```

Comment:
The player object is always initialized, this bug stems from `Game.all` sometimes returning games, other times doesn't.  This test will always fail unless it's run after a test that creates a game, i.e. if this test is run first or on its own, it will always fail. By default, Minitest randomizes test order—which is why you're seeing this error intermittently.

If we force Minitest to only run the one test: `ruby table_foosball_test.rb -n /test_create_new_player/`, you'll see that it always fails because the `Game.all` set is empty. I would solve this by defining `Game.all` as:

```ruby
def self.all
  @games || []
end
```